src/avatar/stories/Avatar.stories.tsx

```diff
@@ -1,153 +1,124 @@
+      <>
+        <Avatar
+          status={isTyping ? "typing" : "online"}
+          className="flex"
+          {...args}
         />
-      </label>
-    </>
-  );
-};
-
-export const Statuses: Story<AvatarProps> = () => (
-  <AvatarGroup size="xl">
-    <Avatar size="xl" src="https://bit.ly/dan-abramov" name="Anurag Hazra">
-      <AvatarBadge>
-        <Status status="online" />
-      </AvatarBadge>
-    </Avatar>
-    <Avatar size="xl" src="https://bit.ly/dan-abramov" name="Anurag Hazra">
-      <AvatarBadge>
-        <Status status="offline" />
-      </AvatarBadge>
-    </Avatar>
-    <Avatar size="xl" src="https://bit.ly/dan-abramov" name="Anurag Hazra">
-      <AvatarBadge>
-        <Status status="sleep" />
-      </AvatarBadge>
-    </Avatar>
-  </AvatarGroup>
-);
-
-export const OnlineTooltip: Story<AvatarProps> = () => {
-  const tooltip = useTooltipState({ placement: "right-start", gutter: 5 });
-
-  return (
-    <Avatar size="xl" name="Anurag Hazra" src="https://bit.ly/dan-abramov">
-      <AvatarBadge>
-        <Tooltip
-          as="div"
-          className="px-1 font-sans text-xs text-white bg-green-500 rounded-full"
-          {...tooltip}
-        >
-          Online
-        </Tooltip>
-        <TooltipReference {...tooltip}>
-          <OnlineDot></OnlineDot>
-        </TooltipReference>
-      </AvatarBadge>
-    </Avatar>
-  );
-};
+        <label className="block mt-4">
+          isTyping?
+          <input
+            type="checkbox"
+            name="typing"
+            className="ml-2"
+            onChange={() => setTyping(!isTyping)}
+          />
+        </label>
+      </>
+    );
+  },
+  { size: "xl", src: "https://bit.ly/dan-abramov" },
+)({});
+
+export const Group = storyTemplate<AvatarProps>(
+  args => (
+    <AvatarGroup {...args}>
+      <Avatar
+        src="https://bit.ly/dan-abramov"
+        name="Dan Abramov"
+        status="online"
+      />
+      <Avatar
+        src="https://bit.ly/kent-c-dodds"
+        name="Kent Dodds"
+        status="sleep"
+      />
+      <Avatar
+        src="https://bit.ly/code-beast"
+        name="Code Beast"
+        status="typing"
+      />
+    </AvatarGroup>
+  ),
+  { size: "xl" },
+)({});
 
-export const Group: Story<AvatarProps> = () => (
+export const GroupWithLimit = () => (
   <>
-    {["xs", "sm", "md", "lg"].map((size, i) => {
+    {["xs", "sm", "md", "lg", "xl"].map((size, i) => {
       return (
-        <>
-          <br />
-          <AvatarGroup limit={i + 1} size={size as any}>
-            <Avatar name="Dan Abrahmov" src="https://bit.ly/dan-abramov" />
-            <Avatar name="Kent Dodds" src="https://bit.ly/kent-c-dodds" />
-            <Avatar name="Ryan Florence" src="https://bit.ly/ryan-florence" />
-            <Avatar name="Christian Nwamba" src="https://bit.ly/code-beast" />
-          </AvatarGroup>
-        </>
+        <AvatarGroup
+          limit={i + 1}
+          className="flex mt-4"
+          size={size as any}
```

Comment:
We can as assertion by doing const assertion on the array

`const sizes = ["xs", "sm", "md", "lg", "xl"] as const;` 

![image](https://user-images.githubusercontent.com/35374649/107647391-2028a080-6ca1-11eb-8cf5-e7a6046e1534.png)